XiEditor/Document.swift

```diff
@@ -38,6 +38,8 @@ class Document: NSDocument {
     // minimum size for a new or resized window
     static var minWinSize = NSSize(width: 240, height: 160)
 
+    var lines = LineCache<LineAssoc>()
```

Comment:
Interestingly, I had better performance in Xi with this change than without it. Opening mod.rs from the Rust source code had no hitches, while master's version would leave me with blank lines on the first few lines.

Reply:
You mean on current master/master you can get blank lines at the start of a document?

Reply:
I thought that was the case, but opening those aforementioned files in master seemed okay. I might have mistaken it with some other branch?

Reply:
yes, I wonder. That was a behaviour I was worried about with this work; it can happen if we miss the first `update`, which includes the first lines. 😕

Reply:
It doesn't seem to happen to me on both my testing machines (a MBP 2014 and a MBP 2015), though perhaps it can show on a slower machine.